pkg/scyllaclient/client.go

```diff
@@ -301,8 +301,16 @@ func (c *Client) HasSchemaAgreement(ctx context.Context) (bool, error) {
 	return len(versions) == 1, nil
 }
 
-func DefaultTransport() *http.Transport {
-	return &http.Transport{
+type TransportOption func(transport *http.Transport)
+
+func WithTLSConfig(tlsConfig *tls.Config) func(*http.Transport) {
```

Comment:
If the field is public and can be set on the returned config, what's the benefit of option wrappers? That's usually done for private fields.

Reply:
Short extensible code. Options pattern not always modify non-exported fields, they are also used for chaning default values.
Your previous [comment](https://github.com/scylladb/scylla-operator/pull/884#discussion_r770325025) suggested writing directly to cfg.Transport which is an interface of http.RoundTripper and cannot be modified, so I added an option to achive the same with shorter block and without additional object allocation.

Reply:
>writing directly to cfg.Transport which is an interface of http.RoundTripper and cannot be modified 

I don't feel like that argument relates to this interface as it's behind the boundary. The function returns `*http.Transport` which has the public field that needs to be set.

this looks much simpler 
```
t := DefaultTransport()
t.TLSClientConfig = myTLSConfig
```
then the lambda syntax
```
func WithTLSConfig(tlsConfig *tls.Config) func(*http.Transport) {
	return func(transport *http.Transport) {
		transport.TLSClientConfig = tlsConfig
	}
}

t := DefaultTransport(WithTLSConfig(myTLSConfig))
```
which also isn't consistent with setting other option which would lack the lambda wrapper or needed manufacturing one per each.

Reply:
> this looks much simpler
> 
> ```
> t := DefaultTransport()
> t.TLSClientConfig = myTLSConfig
> ```

suggested change was implemented in first version and received a remark to make it shorter. I'll revert to initial version then.

> which also isn't consistent with setting other option which would lack the lambda wrapper or needed manufacturing one per each.

It's a tradeoff of keeping dead code in internal library or having consistency and option to modify all possible parameters.

Reply:
> suggested change was implemented in first version and received a remark to make it shorter.

apologies, if we have misunderstood each other there